src/bindings/python/src/pyopenvino/core/common.cpp

```diff
@@ -194,7 +195,7 @@ void fill_tensor_from_strings(ov::Tensor& tensor, py::array& array) {
             PyUnicode_FromKindAndData(PyUnicode_4BYTE_KIND, reinterpret_cast<void*>(ptr), buf.itemsize / 4);
         PyObject* _utf8_obj = PyUnicode_AsUTF8String(_unicode_obj);
         const char* _tmp_str = PyBytes_AsString(_utf8_obj);
-        data[i] = std::string(_tmp_str);
+        data[i] = std::string(_tmp_str, buf.strides[0]);
```

Comment:
Maybe the number of characters should be taken from `buf` object like at [L181](https://github.com/openvinotoolkit/openvino/pull/26121/files#diff-622a251f061d9a537d2f4a087dc8e5666ce22fe246abc7cf69c0b31e5abb9870R181) ?